### PR TITLE
[Accessibility] add automated accessibility CI checks of core features

### PR DESCRIPTION
**Pull request by emmaindal**
## Hey, I just made a Pull Request!

In this PR we add a new Lighthouse Github Action that will run on changes to any of the core features (including, TechDocs, Search, Software Catalog and Software Templates). The check will pass if its over a score of `1` (max) and fail if not. The output will look like:

```
Checking assertions against 11 URL(s), 11 total run(s)

1 result(s) for http://localhost:3000/catalog :

  ✘  categories.accessibility failure for minScore assertion
        expected: >=1
           found: 0.98
      all values: 0.98

1 result(s) for http://localhost:3000/catalog/default/component/backstage :

  ✘  categories.accessibility failure for minScore assertion
        expected: >=1
           found: 0.97
      all values: 0.97

1 result(s) for http://localhost:3000/docs :

  ✘  categories.accessibility failure for minScore assertion
        expected: >=1
           found: 0.98
      all values: 0.98

1 result(s) for http://localhost:3000/docs/default/component/backstage :

  ✘  categories.accessibility failure for minScore assertion
        expected: >=1
           found: 0.97
      all values: 0.97

1 result(s) for http://localhost:3000/create :

  ✘  categories.accessibility failure for minScore assertion
        expected: >=1
           found: 0.86
      all values: 0.86

1 result(s) for http://localhost:3000/create/tasks :

  ✘  categories.accessibility failure for minScore assertion
        expected: >=1
           found: 0.9
      all values: 0.9

1 result(s) for http://localhost:3000/create/actions :

  ✘  categories.accessibility failure for minScore assertion
        expected: >=1
           found: 0.98
      all values: 0.98

1 result(s) for http://localhost:3000/create/edit :

  ✘  categories.accessibility failure for minScore assertion
        expected: >=1
           found: 0.[98](https://github.com/backstage/backstage/actions/runs/4609299256/jobs/8146249757?pr=17242#step:5:99)
      all values: 0.98
```

where one can see what paths are failing. To produce the more detailed reports, one can run the tests locally using the lighthouse CLI

```
npm install -g @lhci/cli@0.11.x
lhci autorun
```
#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [ ] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: lighthouserc.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+      puppeteerScript: './.lighthouseci/scripts/guest-auth.js',
+      startServerReadyPattern: 'Listening on :7007',
+    },
+    assert: {
+      assertions: {
+        'categories:performance': 'off',
+        'categories:pwa': 'off',
+        'categories:best-practices': 'off',
+        'categories:seo': 'off',
+        'categories:accessibility': ['error', { minScore: 1 }],
```
Comment (emmaindal) [inline lighthouserc.js:56]: we probably don't want to set this to `1` yet. But as soon as all issues has been resolved in core features.

Comment (emmaindal) [reply lighthouserc.js]: lowered score to 0.85 here https://github.com/backstage/backstage/pull/17242/commits/ba544c227db30e38be7f842a82235e4225419576 as we have follow up issues to resolve. E.g. #17279.

**Review by aitherios — APPROVED** [review]

**Comment by freben** [comment]
Hi! You should have a changeset for the search plugin. surprisingly - the readme is part of the published data

**Comment by emmaindal** [comment]
@freben well, then lets skip the search docs update as part of this PR, it was mostly to trigger the check, nothing urgent and I'll make those updates when we release a new version for search around something else

File: .github/workflows/verify_accessibility_core.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Use Node.js 16.x
+        uses: actions/setup-node@v3
+        with:
+          node-version: 16.x
+      - name: install dependencies
+        run: |
+          yarn install
```
Comment (Rugvip) [inline .github/workflows/verify_accessibility_core.yml:23]:
```suggestion
      - name: yarn install
        uses: backstage/actions/yarn-install@v0.6.3
        with:
          cache-prefix: ${{ runner.os }}-16.x
```

Could use the common action for this right? keeping things a bit speedy

**Review by freben — APPROVED** [review]
Let's go!

**Pull request merged**